gateway-ha/src/main/java/io/trino/gateway/ha/security/LbFilter.java

```diff
@@ -13,32 +13,45 @@
  */
 package io.trino.gateway.ha.security;
 
-import io.dropwizard.auth.AuthFilter;
+import io.trino.gateway.ha.security.util.AuthenticationException;
+import io.trino.gateway.ha.security.util.Authorizer;
+import io.trino.gateway.ha.security.util.IdTokenAuthenticator;
 import jakarta.annotation.Nullable;
 import jakarta.annotation.Priority;
-import jakarta.ws.rs.Priorities;
 import jakarta.ws.rs.WebApplicationException;
 import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
 import jakarta.ws.rs.core.Cookie;
 import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.SecurityContext;
 
 import java.security.Principal;
 import java.util.Optional;
 
-@Priority(Priorities.AUTHENTICATION)
-public class LbFilter<P extends Principal>
-        extends AuthFilter<String, P>
+import static jakarta.ws.rs.Priorities.AUTHENTICATION;
+import static java.util.Objects.requireNonNull;
+
+@Priority(AUTHENTICATION)
+public class LbFilter
```

Comment:
for all these classes I think you are implying lb = Load Balancer = Trino Gateway .. right? If so ... I am wondering if we really want that ... our naming is already a bit all over the place .. and introducing lb when we already have gateway and hagateway and Trino Gateway in use seems bad

Reply:
However .. we should fix that in another PR later since this is already in the current codebase..

Reply:
I'd like to keep the original naming during refactoring. Otherwise it'll mess up the diff.